fauna/fauna-shell
# FE-818 | Prompt user for OTP when MFA is enabled on account

### Notes
[Jira Ticket](https://faunadb.atlassian.net/browse/FE-818)

Now that we've enabled MFA in Dashboard, we want to make sure we're protecting data that could be accessed via Fauna Shell. Whenever a user runs `cloud-login` but has MFA enabled on their account, they will now be prompted to enter their OTP code before successfully authenticating.

### How to test
~~1. Pull down this branch
2. Set up a user on https://dashboard.fauna-preview.com/ and enable MFA
3. Back in your local shell, run `bin/run cloud-login`
4. You should be prompted for a username, password and an OTP code
5. Upon success, run `bin/run shell --endpoint [NAME]` where `name` is whatever you called the endpoint in Step 3
6. Run some FQL queries to confirm everything is working~~

### How to test locally
1. Run your local `dashboard`, `auth-service` and `db`
2. Login to the Dashboard, create a new user, enable MFA and scan the QR code
3. In `cloud-login.js`, you'll want to make the changes below:
```javascript
const SHELL_LOGIN_URL = process.env.FAUNA_SHELL_LOGIN_URL
  ? process.env.FAUNA_SHELL_LOGIN_URL
  : "http://localhost:3000/login";

const CLOUD_URL = "http://localhost:8443";
```
4. In the repo directory, run `bin/run cloud-login`. You should be prompted for a email, password and OTP code (since you enabled MFA in step 2)
5. Now, run `bin/run shell --endpoint cloud` and run a few queries to confirm everything is working


### Screencap
https://www.loom.com/share/183072399cbb4e6890701fb97e627ace
